app/models/theme_modifier_set.rb

```diff
@@ -75,14 +75,39 @@ def topic_thumbnail_sizes=(val)
     super(val.map { |dim| "#{dim[0]}x#{dim[1]}" })
   end
 
+  def add_theme_setting_modifier(modifier_name, setting_name)
+    self.theme_setting_modifiers ||= {}
+    self.theme_setting_modifiers[modifier_name] = setting_name
+  end
+
+  def refresh_theme_setting_modifiers(target_setting_name: nil, target_setting_value: nil)
+    changed = false
+    if self.theme_setting_modifiers.present?
+      self.theme_setting_modifiers.each do |modifier_name, setting_name|
+        modifier_name = modifier_name.to_sym
+        setting_name = setting_name.to_sym
+
+        next if target_setting_name.present? && target_setting_name.to_sym != setting_name
+
+        value =
+          target_setting_name.present? ? target_setting_value : theme.settings[setting_name]&.value
+        if read_attribute(modifier_name) != value
```

Comment:
We need to think about how this works for arrays (including multi-dimensional-arrays, like 'topic_thumbnail_sizes'). Theme modifiers have control over their database storage type. Theme settings are just stored as strings.

Reply:
had to close this cause I messed up git, will make a new pr